Add missing PropTypes for components

## Context
Currently, we have two files where we are missing prop types (Resources/ResourceCard.js and Resources/ResourcePage.js). These can be found by running `npm run lint` and it should show what is currently missing. At the time of this writing, the missing prop types are:

```bash
/react-concept/src/components/Resources/ResourceCard.js
  47:32  warning  'id' is missing in props validation           react/prop-types
  47:36  warning  'title' is missing in props validation        react/prop-types
  47:43  warning  'created' is missing in props validation      react/prop-types
  47:52  warning  'description' is missing in props validation  react/prop-types
  47:65  warning  'url' is missing in props validation          react/prop-types

/react-concept/src/components/Resources/ResourcePage.js
   8:25  warning  'matchProps' is missing in props validation                  react/prop-types
  13:60  warning  'matchProps.match' is missing in props validation            react/prop-types
  13:66  warning  'matchProps.match.params' is missing in props validation     react/prop-types
  13:73  warning  'matchProps.match.params.id' is missing in props validation  react/prop-types
  23:18  warning  'matchProps.match' is missing in props validation            react/prop-types
  23:24  warning  'matchProps.match.params' is missing in props validation     react/prop-types
  23:31  warning  'matchProps.match.params.id' is missing in props validation  react/prop-types
```

## Implementation
1. Run `npm run lint` and find out which components have missing prop types
1. Address missing prop types
1. Change the rules in the eslintrc file so that missing prop types return an error instead of a warning

## Acceptance criteria
- [x] No more missing prop types
- [x] ESLint rule is updated and is working correctly

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -24,7 +24,7 @@
   "plugins": ["react", "prettier"],
   "rules": {
     "prettier/prettier": "error",
-    "react/prop-types": "warn",
+    "react/prop-types": "error",
     "react/no-unescaped-entities": 0,
     "comma-dangle": ["error", "only-multiline"],
     "no-unused-vars": [2, { "vars": "all", "args": "after-used" }]

--- a/src/components/Resources/ResourceCard.js
+++ b/src/components/Resources/ResourceCard.js
@@ -1,4 +1,5 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import { makeStyles } from '@material-ui/core/styles';
 import { red } from '@material-ui/core/colors';
 import { Link } from 'react-router-dom';
@@ -117,4 +118,12 @@ export const ResourceCard = ({ id, title, created, description, url }) => {
       </Collapse>
     </Card>
   );
+};
+
+ResourceCard.propTypes = {
+  id: PropTypes.number,
+  title: PropTypes.string,
+  created: PropTypes.string,
+  description: PropTypes.string,
+  url: PropTypes.string,
 };

--- a/src/components/Resources/ResourcePage.js
+++ b/src/components/Resources/ResourcePage.js
@@ -1,4 +1,5 @@
 import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
 import axios from 'axios';
 import PersonalMenu from '../PersonalMenu';
@@ -45,5 +46,9 @@ function ResourcePage({ matchProps }) {
     </>
   );
 }
+
+ResourcePage.propTypes = {
+  matchProps: PropTypes.object,
+};
 
 export default ResourcePage;

